packages/js-lib/src/zod/zod.shared.schemas.test.ts

```diff
@@ -0,0 +1,57 @@
+import { getAjv } from '@naturalcycles/nodejs-lib'
+import { describe, expect, test } from 'vitest'
+import { customZodSchemas, z } from './index.js'
+
+test.each(Object.keys(customZodSchemas))(
+  'custom zod schemas like "z.%s" should properly convert to JSON schema and AJV schema',
+  key => {
+    const validator = customZodSchemas[key as keyof typeof customZodSchemas]
+    const zodSchema = z.object({ value: validator })
+    const jsonSchema = z.toJSONSchema(zodSchema, { target: 'draft-7' })
+    const ajvSchema = getAjv().compile(jsonSchema)
+
+    expect(ajvSchema).toBeDefined()
+  },
+)
+
+describe('z.email', () => {
+  test('should accept a valid email address', () => {
+    const email = 'test@example.com'
+    const result = z.email.parse(email)
```

Comment:
I noticed a difference in syntax - zod default schemas are functions, etc `number()`, and here you're introducing `email` as a property (not a function).

I suggest to have all schemas as functions, to be consistent with zod.

Why do I believe they are functions? So they can take options, for configuration.

Reply:
> I cannot recall any JOI schema that we parametrized

There are a few, e.g `dateString` takes a `'today'` parameter (like `dateString('today')`), `arraySchema(something)`. Most are static.

My main point was: consistency with how zod defines things, e.g `z.number()` as a function, vs `z.number` as a property